iroh-net/src/endpoint.rs

```diff
@@ -296,17 +293,41 @@ impl Builder {
     }
 }
 
+/// Configuration for a [`quinn::Endpoint`] that cannot be changed at runtime.
+#[derive(Debug)]
+struct StaticConfig {
+    secret_key: SecretKey,
+    transport_config: Arc<quinn::TransportConfig>,
+    keylog: bool,
+    concurrent_connections: Option<u32>,
+}
+
+impl StaticConfig {
+    /// Build a [`quinn::ServerConfig`] with the specified ALPN protocols.
+    fn build(&self, alpn_protocols: Vec<Vec<u8>>) -> Result<quinn::ServerConfig> {
```

Comment:
I don't like the name. It sounds like it builds a StaticConfig, but it does build a final ServerConfig.

Reply:
what if we renamed `StaticConfig` to `ServerConfigBuilder`?

Reply:
I went with `StaticConfig::create_server_config` for now. I'm fine to change further, don't mind much (and it's not pubic API, internal only).